### PR TITLE
#41 - Fix issue causing getters to be invoked when copying over to the React component

### LINKED ISSUE #41: Copying prototyped methods / properties over to the react component invokes the getter
Copying over prototype members from a web component invokes the getter. If any code executed in the getter requires an instance be set up, things can break. For example:

```js
get lcTagName() {
  return this.tagName.toLowerCase();
}
```

That will cause an error because `tagName` will either not be invokable from a DOM node that hasn't been instantiated, or the prototype may not be set up (in polyfills).

### PR DESCRIPTION
**Pull request by treshugart**
Also fixed to copy over non-enumerable properties.

Fixes #41

### CODE REVIEW

File: test/unit/webcomponent-proto-funcs.js [hunk 1]
```diff
@@ -1,32 +1,44 @@
...
-  };
-  proto.getProp = function() {
-    return this.prop;
-  };
   return reactify(document.registerElement(`x-webcomponent-proto-funcs-${x++}`, {
-    prototype: proto,
-  }), { React, ReactDOM });
+    prototype: Object.create(HTMLElement.prototype, {
+      prop: {
+        value: 'prop',
```
Comment (bradleyayers) [inline test/unit/webcomponent-proto-funcs.js]: Where does this `value` stuff come from?

Comment (treshugart) [inline test/unit/webcomponent-proto-funcs.js]: `Object.defineProperty()`

Comment (lukebatchelor) [inline test/unit/webcomponent-proto-funcs.js]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/defineProperty

Comment (lukebatchelor) [inline test/unit/webcomponent-proto-funcs.js]: I think its that. How you set up properties on a prototype

**Comment by bradleyayers** [comment]
Seems reasonable, lgtm.

**Comment by stevemao** [comment]
lgtm

We can probably remove `approvals/lgtm` since GitHub supports reviews.

**Comment by treshugart** [comment]
@stevemao yeah, I've already removed them. I think it's just still there because I removed it after submitting this PR.

**Pull request merged**